### PR TITLE
Fix parameters in named collection for PostgreSQL

### LINKED ISSUE #60936: Broken postgres dictionary
Server version: 23.9. Also tested on fiddle with latest

Dictionary with POSTGRESQL engine using named collection not workin

Creating named collection
`create  named collection pgsql as user = 'clickhouse', password = 'pass', database = 'db', host = 'host.host', port = '5501', schema = 'public'`
then creating dictionary

```
 CREATE DICTIONARY db.dictionary
(
    `id` UInt64,
    `ip` String,
    `name` String,
    `comment` String,
    `type` UInt64
)
PRIMARY KEY ip
SOURCE(POSTGRESQL(
   NAME pgsql
   QUERY "
   <some_query>
   "
))
LIFETIME(MIN 0 MAX 3600)
LAYOUT(IP_TRIE)
```
When trying to reload or use dictionary
```
Received exception from server (version 23.9.6):
Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: There is no collection named `pgsql` in config. (BAD_ARGUMENTS)
```
And in err.lof
```
2024.03.06 15:24:12.833197 [ 1984093 ] {eb843e1d-2056-4767-83f0-c9b56191d331} <Error> TCPHandler: Code: 36. DB::Exception: There is no collection named `pgsql` in config. (BAD_ARGUMENTS), Stack trace (when copying this message, always include the lines below):

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000c73ac37 in /usr/bin/clickhouse
1. DB::Exception::Exception<String&>(int, FormatStringHelperImpl<std::type_identity<String&>::type>, String&) @ 0x00000000075a24cd in /usr/bin/clickhouse
2. std::optional<DB::ExternalDataSourceInfo> DB::getExternalDataSourceConfiguration<DB::EmptySettingsTraits>(Poco::Util::AbstractConfiguration const&, String const&, std::shared_ptr<DB::Context const>, std::function<bool (String const&)>, DB::BaseSettings<DB::EmptySettingsTraits> const&) @ 0x0000000012793616 in /usr/bin/clickhouse
3. DB::getExternalDataSourceConfigurationByPriority(Poco::Util::AbstractConfiguration const&, String const&, std::shared_ptr<DB::Context const>, std::function<bool (String const&)>) @ 0x0000000012798452 in /usr/bin/clickhouse
4. std::shared_ptr<DB::IDictionarySource> std::__function::__policy_invoker<std::shared_ptr<DB::IDictionarySource> (DB::DictionaryStructure const&, Poco::Util::AbstractConfiguration const&, String const&, DB::Block&, std::shared_ptr<DB::Context const>, String const&, bool)>::__call_impl<std::__function::__default_alloc_func<DB::registerDictionarySourcePostgreSQL(DB::DictionarySourceFactory&)::$_0, std::shared_ptr<DB::IDictionarySource> (DB::DictionaryStructure const&, Poco::Util::AbstractConfiguration const&, String const&, DB::Block&, std::shared_ptr<DB::Context const>, String const&, bool)>>(std::__function::__policy_storage const*, DB::DictionaryStructure const&, Poco::Util::AbstractConfiguration const&, String const&, DB::Block&, std::shared_ptr<DB::Context const>&&, String const&, bool) (.llvm.17243121882007358706) @ 0x000000000f4eec37 in /usr/bin/clickhouse
5. DB::DictionarySourceFactory::create(String const&, Poco::Util::AbstractConfiguration const&, String const&, DB::DictionaryStructure const&, std::shared_ptr<DB::Context const>, String const&, bool) const @ 0x0000000010d3c473 in /usr/bin/clickhouse
6. DB::DictionaryFactory::create(String const&, Poco::Util::AbstractConfiguration const&, String const&, std::shared_ptr<DB::Context const>, bool) const @ 0x0000000010d38db3 in /usr/bin/clickhouse
7. DB::ExternalDictionariesLoader::create(String const&, Poco::Util::AbstractConfiguration const&, String const&, String const&) const @ 0x0000000011cb32bc in /usr/bin/clickhouse
8. DB::ExternalLoader::LoadingDispatcher::doLoading(String const&, unsigned long, bool, unsigned long, bool, std::shared_ptr<DB::ThreadGroup>) @ 0x0000000011cbe537 in /usr/bin/clickhouse
9. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<void (DB::ExternalLoader::LoadingDispatcher::*)(String const&, unsigned long, bool, unsigned long, bool, std::shared_ptr<DB::ThreadGroup>), DB::ExternalLoader::LoadingDispatcher*, String&, unsigned long&, bool&, unsigned long&, bool, std::shared_ptr<DB::ThreadGroup>>(void (DB::ExternalLoader::LoadingDispatcher::*&&)(String const&, unsigned long, bool, unsigned long, bool, std::shared_ptr<DB::ThreadGroup>), DB::ExternalLoader::LoadingDispatcher*&&, String&, unsigned long&, bool&, unsigned long&, bool&&, std::shared_ptr<DB::ThreadGroup>&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x0000000011cc35ff in /usr/bin/clickhouse
10. void* std::__thread_proxy[abi:v15000]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000c824067 in /usr/bin/clickhouse
11. start_thread @ 0x00000000000081ca in /usr/lib64/libpthread-2.28.so
12. clone @ 0x0000000000039e73 in /usr/lib64/libc-2.28.so
```
Creating table with engine POSTGRESQL or manual function postgresql() working as expected

A link to reproducer in [https://fiddle.clickhouse.com/a31868af-91d7-49d8-9db7-bdb3d99cbe3b](https://fiddle.clickhouse.com/a31868af-91d7-49d8-9db7-bdb3d99cbe3b).

**Does it reproduce on the most recent release?**

Reproduced with fiddle on 24.2

**Additional context**

> Postgres version 16.1

### PR DESCRIPTION
**Pull request by evillique**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix named collection not being found in parameters for PostgreSQL. Fixes https://github.com/ClickHouse/ClickHouse/issues/60936

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit f4bf15db0f753185977481700c41757e6a4c2fa8 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62311/f4bf15db0f753185977481700c41757e6a4c2fa8/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by jsc0218 — APPROVED** [review]

**Review by kssenii — COMMENTED** [review]
Could you please add a test?

**Comment by patryk4815** [comment]
Any update :)?

**Comment by jsc0218** [comment]
Kindly ping @evillique.

File: src/Storages/ExternalDataSourceConfiguration.cpp
Comment (kssenii) [inline src/Storages/ExternalDataSourceConfiguration.cpp:1]: I deleted this file `ExternalDataSourceConfiguration` completely here https://github.com/ClickHouse/ClickHouse/pull/68412. So let's close this PR as the issue will be automatically fixed in the linked PR?

**Pull request closed without merging**